### DISCUSSION TITLE
How to use request payload on RLS insert check?

### DISCUSSION THREAD

**Discussion by guizoxxv** [supabase/supabase#12853] [Questions]
I want to use a request attribute inside a select on a policy check expression. For example, in my app I have:

* users (id, name)
* teams (id, name, user_id)
* items (id, name, team_id)

A users belong to many teams and it can only create items from teams they belong. Considering the following request to create an item:

```json
{ "name": "Item 1", "team_id": "81a7939d-47bd-430a-aa5e-90db44bc9b4f" }
```

I want to check that the logged user belongs to team in the payload. I guess I could use something like this in the `check` of the insert policy:

```sql
auth.uid() IN (
   SELECT user_id
   FROM teams
   WHERE id = TEAM_ID_FROM_PAYLOAD
)
```
But, how can I use the value from `team_id` in the payload on TEAM_ID_FROM_PAYLOAD?

Thank you.

**Comment by GaryAustin1** [comment]
You should just be able to use  `WHERE id = team_id`

**Reply by guizoxxv** [reply]
In the `WITH CHECK expression`, if I provide:

```sql
auth.uid() IN (
   SELECT user_id
   FROM teams
   WHERE id = team_id
)
```

When I click `edit` on the policy I see this:

```sql
(auth.uid() IN ( SELECT teams.user_id
   FROM teams
  WHERE (teams.id = items.team_id)))
```

It seems the it interprets that I'm not refering to `team_id` from the payload, but from the `items` table, in which I'm executing the insert.

**Reply by GaryAustin1** [reply]
Isn't the table the policy is on `items`?  If so that is just clarifying it is on the `items` table versus the `teams` table that the select is on..  Should work the same as `team_id` in your case.

**Reply by guizoxxv** [reply]
The policy is on the `items` table, but I need to verify that the `team_id` provided in the payload belongs to the logged user, that is, the logged user was the one who created this team. This information will be in the `teams` table, `user_id` column.

**Reply by GaryAustin1** [reply]
This is an insert?  If so the payload is the row being inserted into items.  items.team_id is the value you provided in the insert request for that row to be inserted.

**Comment by guizoxxv** [comment]
Actually, it seems that the issue is not on the policy. I kept the scenario to a minimum to keep it simple but in the actual application there is more.

I'm calling the [signUp](https://supabase.com/docs/reference/javascript/auth-signup) function before inserting. That is because `items` also has a `user_id` column. In one action, a `admin` user (in my app, not on supabase) creates a new normal `user` (signUp) and a `item` (insert into items) that belong to this `user` and a particular `team`.

I believe that this leads me into [this scenario](https://github.com/supabase/supabase/discussions/11909) and causes the policy check to fail.
